src/middlewared/middlewared/plugins/boot.py

```diff
@@ -172,6 +176,11 @@ async def get_scrub_interval(self):
         """
         return (await self.middleware.call('system.advanced.config'))['boot_scrub']
 
+    @private
+    async def handle_initrd_zfs(self, update_initramfs_if_changes=True):
+        await run('/usr/local/bin/initrd-zfs.py', BOOT_POOL_NAME, '/', str(int(update_initramfs_if_changes)),
+                  encoding='utf8', errors='ignore')
```

Comment:
Seems like we could capture output and log if anything wrong happens with that script?

Reply:
This is done automatically, unhandled `subprocess.CalledProcessError` is re-raised as CallError with failed command's output